internal/test-framework/pkg/runner/container/container.go

```diff
@@ -0,0 +1,285 @@
+	cr.Container.Entrypoint = []string{"/bin/bash"}
+	cr.Container.Cmd = []string{"-c", "sleep 10000"}
+
+	// Create container, attach it to the integration network, and wait for completion
+	resp, err := cr.Client.ContainerCreate(ctx, cr.Container, cr.Host, nil, "")
+	if err != nil {
+		return err
+	}
+
+	if err = cr.Client.NetworkConnect(ctx, networks[0].ID, resp.ID, nil); err != nil {
+		return err
+	}
+
+	if err = cr.Client.ContainerStart(ctx, resp.ID, types.ContainerStartOptions{}); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Name discovers the runner container and returns the name of the container.
+func (cr *ContainerRunner) Name(ctx context.Context) (string, error) {
+	containers, err := cr.findRunnerContainer(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	if len(containers) > 1 {
+		return "", fmt.Errorf("multiple containers found matching %s=%s", "talos.integration.name", cr.ClusterName)
+	}
+
+	if len(containers) == 0 {
+		return "", nil
+	}
+
+	// No idea why Names is []string
+	return containers[0].Names[0], nil
+}
+
+// exists checks to see if there is only a single runner container available.
+// If more than one container is returned from our search, we error. Otherwise
+// we'll use the container if it exists or create a new one if it does not.
+func (cr *ContainerRunner) exists(ctx context.Context) (bool, error) {
+	containers, err := cr.findRunnerContainer(ctx)
+	if err != nil {
+		return false, err
+	}
+
+	if len(containers) > 1 {
+		return false, err
+	}
+
+	if len(containers) == 0 {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+// findRunnerContainer discovers the container to be used for command execution
+// by searching through labels `talos.owned` and `talos.integration.name`.
+func (cr *ContainerRunner) findRunnerContainer(ctx context.Context) ([]types.Container, error) {
+	// Test if integration container already exists
+	containerFilters := filters.NewArgs()
+	containerFilters.Add("label", fmt.Sprintf("%s=%s", "talos.owned", "true"))
+	containerFilters.Add("label", fmt.Sprintf("%s=%s", "talos.integration.name", cr.ClusterName))
+
+	return cr.Client.ContainerList(ctx, types.ContainerListOptions{Filters: containerFilters})
+}
+
+// prepareExec ensures that the runner container is running.
+func (cr *ContainerRunner) prepareExec(ctx context.Context) (string, error) {
+	err := cr.start(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	return cr.Name(ctx)
+}
+
+// runCommandFn sets up the actual command to be run inside the container.
+func (cr *ContainerRunner) runCommandFn(ctx context.Context, check *checker.Check, container string) func() error {
+	return func() error {
+		var (
+			err     error
+			idresp  types.IDResponse
+			inspect types.ContainerExecInspect
+			resp    types.HijackedResponse
+		)
+
+		execConfig := types.ExecConfig{
+			Tty:          true,
+			AttachStdin:  true,
+			AttachStdout: true,
+			AttachStderr: true,
+			Cmd:          []string{"/bin/bash", "-c", check.Command.String()},
```

Comment:
as a suggestion, we can prepend safety defaults: https://github.com/talos-systems/bldr/blob/master/internal/pkg/types/v1alpha2/instruction.go#L15